pvanalytics/quality/irradiance.py

```diff
@@ -471,3 +471,174 @@ def daily_insolation_limits(irrad, clearsky, daily_min=0.4, daily_max=1.25):
         lower_bound=daily_min
     )
     return good_days.reindex(irrad.index, method='pad', fill_value=False)
+
+
+def _fill_nighttime(component, component_sum_df,
+                    fill_nighttime, fill_value, solar_zenith, zenith_limit):
+    # This function is used to fill in nighttime values for the computed
+    # irradiance time series (GHI, DHI, DNI).
+    # Set the series based on the component.
+    if component == 'GHI':
+        series = component_sum_df['ghi']
+    elif component == 'DHI':
+        series = component_sum_df['dhi']
+    else:
+        series = component_sum_df['dni']
+    # Logic for filling in nighttime values for a
+    # component sum series.
+    # Find the locations where the sun is below the sza limit.
+    mask = (zenith_limit <= solar_zenith)
+    if fill_nighttime == 'fill_value':
+        # Replace the nighttime values with a fill value
+        series[mask] = fill_value
+    if fill_nighttime == 'equation':
+        # Use the nighttime equation GHI = 0 + DHI, which translates as:
+        # GHI_Calc (at night) = DHI_measured
+        # DHI_Calc (at night) = GHI_measured
+        # DNI_Calc (at night) = 0
+        if component == 'GHI':
+            series[mask] = component_sum_df['dhi'][mask]
+        elif component == 'DHI':
+            series[mask] = component_sum_df['ghi'][mask]
```

Comment:
Are these reversed?

Reply:
This is the intended implementation as the equation simplifies from:
GHI = DHI + DNI * cos(theta_z)
To:
 GHI = 0 + DHI
at nighttime (as DNI=0). By this logic, the GHI data is equal to the DHI data (and vice versa) at nighttime.

Reply:
Ah, I understand. `component` is the Series to be filled.